doc/python/domdec.rst

```diff
@@ -38,6 +40,64 @@ describes the cell groups on the local MPI rank.
         computational cost, hence it may not produce a balanced partition for
         models with cells that have a large variance in computational costs.
 
+.. function:: partition_by_group(recipe, context, groups)
+
+   Construct a :class:`domain_decomposition` that assigns the groups described by
+   the provided list of :class:`group_description` to the local hardware of the calling rank.
```

Comment:
It would be helpful to provide a list of constraints here to be upheld:
- GJ connections force cells into one group
- a group must be homogeneous in its cell_kind
- ...

Reply:
Yes, I'll add the GJ connections rule. The groups are homogeneous in cell-kind by design.

Reply:
Yes, but it's a constraint that needs to be observed by eg partition_load_balance and thus I would suspect a reader might be interested in that fact.